### PR TITLE
FLOAT support for hive (complete) (2)

### PR DESCRIPTION
**Pull request by cawallin**
Revision of #229 

Includes a fix by @losipiuk that unquarantines TestAllDatatypesFromHiveConnector, and is rebased on the prestodb/presto PR for FLOAT coercions

Squashes all the Hive changes to one commit.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/type/TypeJsonUtils.java [hunk 1]
```diff
@@ -284,6 +287,9 @@ else if (javaType == long.class) {
             if (element instanceof SqlDecimal) {
                 type.writeLong(blockBuilder, ((SqlDecimal) element).getUnscaledValue().longValue());
             }
+            else if (FLOAT.equals(type)) {
+                type.writeLong(blockBuilder, floatToRawIntBits(((Number) element).floatValue()));
```
Comment (maciejgrzybek) [inline presto-main/src/main/java/com/facebook/presto/type/TypeJsonUtils.java:290]: Can we add a test for it?

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java [hunk 2]
```diff
@@ -157,26 +159,29 @@ else if (type.equals(BIGINT) && (statistics instanceof LongStatistics || statist
...
+        else if (type.equals(DOUBLE) && statistics instanceof DoubleStatistics) {
+            DoubleStatistics doubleStatistics = (DoubleStatistics) statistics;
+            // ignore corrupted statistics
+            if (doubleStatistics.genericGetMin() > doubleStatistics.genericGetMax()) {
+                return Domain.create(ValueSet.all(type), hasNullValue);
             }
+            ParquetDoubleStatistics parquetDoubleStatistics = new ParquetDoubleStatistics(doubleStatistics.genericGetMin(), doubleStatistics.genericGetMax());
             return createDomain(type, hasNullValue, parquetDoubleStatistics);
         }
+        else if (type.equals(FLOAT) && statistics instanceof FloatStatistics) {
```
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:171]: Shouldn't we maybe log a warning if for type Float we've got statistics of different type?

**Comment by maciejgrzybek** [comment]
Shouldn't bd7186a78f47dee973f0b4b74ced2f6ea463cd64 be squashed with previous commit?
Looks like it's not something super specific and we have a lot of changes in bd7186a78f47dee973f0b4b74ced2f6ea463cd64^ anyway. ;)

File: presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java [hunk 3]
```diff
@@ -127,6 +127,12 @@ public static void setTypeInfo(Builder builder, TypeSignature type)
                 builder.setScale(0);
                 builder.setColumnDisplaySize(24);
                 break;
+            case "float":
+                builder.setSigned(true);
+                builder.setPrecision(7);
```
Comment (maciejgrzybek) [inline presto-jdbc/src/main/java/com/facebook/presto/jdbc/ColumnInfo.java:132]: Precision is 9 (not for all number though), e.g. `147483648f` is representable in IEEE754 float.

File: presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java [hunk 4]
```diff
@@ -222,6 +222,7 @@ public void testTypes()
                         ", DATE '2013-03-22' as g" +
                         ", INTERVAL '123-11' YEAR TO MONTH as h" +
                         ", INTERVAL '11 22:33:44.555' DAY TO SECOND as i" +
+                        ", FLOAT '123.45' as j" +
```
Comment (maciejgrzybek) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java:225]: We should add here `Float 'Infinity'` as well.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 5]
```diff
@@ -207,7 +207,7 @@ private void assertColumnTypes(QueryResult queryResult)
                 SMALLINT,
                 INTEGER,
                 BIGINT,
-                DOUBLE,
+                REAL,
```
Comment (maciejgrzybek) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java:210]: Maybe make a comment _JDBC.REAL = FLOAT_ or similar?

**Comment by maciejgrzybek** [comment]
A few small comments but in general looks good.

File: presto-main/src/main/java/com/facebook/presto/type/TypeJsonUtils.java (hunk 1 above)
Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/type/TypeJsonUtils.java:290]: It was caught by the current Hive tests, but I'll look to see if there are any more specific unit tests we can add.

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java (hunk 2 above)
Comment (cawallin) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:171]: None of the other types do that, so I'm hesitant to add it. And if you have the wrong type of statistics then you have corrupted statistics of some sort, so it's reasonable to not do pushdown.

**Comment by cawallin** [comment]
@maciejgrzybek Well, bd7186a is more general than the Hive FLOAT patch, so it made sense to separate it, but I can merge it if you think it better.

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java (hunk 2 above)
Comment (maciejgrzybek) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/TupleDomainParquetPredicate.java:171]: Definitely it's not reasonable to pushdown then but it's probably also reasonable to notify user about such a case.
But you're right. Even if it's worth having such a log message, it should be a separate change for all types, not a part of FLOAT PR.

**Pull request closed without merging**